pkg/build/api/types.go

```diff
@@ -70,6 +70,20 @@ type BuildSpec struct {
 	// scheduled in the system, that the build may be active on a node before the
 	// system actively tries to terminate the build; value must be positive integer
 	CompletionDeadlineSeconds *int64
+	// Optional field that holds information about a builds' trigger.
+	BuildTriggerInfo BuildTriggerInfo
+}
+
+// BuildTriggerInfo is an optional field that stores information abou a
+// triggered build.
+type BuildTriggerInfo struct {
+	User                    string
+	BuildTriggerType        string
+	BuildTriggerInformation string
```

Comment:
What information will be stored here? Shouldn't this be `Reason BuildTriggerReason` ?